app/mailers/spree/order_mailer_decorator.rb

```diff
@@ -6,8 +6,8 @@
   def cancel_email(order, resend = false)
     @order = find_order(order)
```

Comment:
but by creating an `order` ivar we could be possibly shadowing any ivar already set in the `OrderMailer` definition (very likely given the name of the class). IMO creating a local var does the job and it is less risky.

We could rename the `order` argument to `order_or_order_id`. Isn't that what is happening?

Reply:
very interesting little comment...
because I dont think your main point is valid, we have to set and keep @order because it's used in the mailer views.
BUT
you made me realise that after this
https://github.com/openfoodfoundation/openfoodnetwork/pull/2589/commits/0de691b8216f5ec183c7098021b6d37cecd2b6ad
we can just remove this cancel_email method :-) and use the spree order_mailer one.
Your comment still applies… but it’s now spree code. I think we will keep (in this case, the not so good) spree code as it is… this in wrt to cancel_email method.

I really liked your order_or_order_id variable name so I decided to use it in the remaining methods (I love making code smells more explicit like this function parameter that takes all types of variations without indication).

As I was refactoring... I also extracted email subject related copy pasted logic to a new build_subject method.

And as I was improving this space, I also removed MailMethod dependency from both this order_mailer and the subscription mailer specs.

I kept the initial commit as it makes historical sense and added 2 new commits with what I mention in this comment.